Automattic/woocommerce-payments
# Descriptive error message when payment method is invalid

Fixes #7301

#### Changes proposed in this Pull Request

At the moment of writing, if a payment method is invalid or not provided, an error message is shown that tells the customer to use a different credit card. This can lead to confusion, especially when customers opt for payment methods other than credit cards. This PR addresses this problem by changing that error message in a generic way.









#### Testing instructions





* The simplest way to get this error is to alter the code here https://github.com/Automattic/woocommerce-payments/blob/ee35b7685964a53db150de342d1f690598f2e736/includes/class-payment-information.php#L134 to trigger the `if` condition.
* Make sure that this error is shown `Invalid or missing payment details. Please ensure the provided payment method is correctly entered.` instead of ` 'Invalid payment method. Please input a new card number.'`

-------------------


**Post merge**
